src/oidc/tokenRefresher.ts

```diff
@@ -60,57 +71,55 @@ export abstract class OidcTokenRefresher {
         deviceId: string,
         redirectUri: string,
     ): Promise<void> {
-        const config = await discoverAndValidateAuthenticationConfig(authConfig);
-
-        const scope = generateScope(deviceId);
-
-        this.oidcClient = new OidcClient({
-            ...config.metadata,
-            client_id: clientId,
-            scope,
-            redirect_uri: redirectUri,
-            authority: config.metadata.issuer,
-            stateStore: new WebStorageStateStore({ prefix: "mx_oidc_", store: window.sessionStorage }),
-        });
+        try {
+            const config = await discoverAndValidateAuthenticationConfig(authConfig);
+
+            const scope = generateScope(deviceId);
+
+            this.oidcClient = new OidcClient({
+                ...config.metadata,
+                client_id: clientId,
+                scope,
+                redirect_uri: redirectUri,
+                authority: config.metadata.issuer,
+                stateStore: new WebStorageStateStore({ prefix: "mx_oidc_", store: window.sessionStorage }),
+            });
+        } catch (error) {
+            logger.error("Failed to initialise OIDC client.", error);
+            throw new Error("Failed to initialise OIDC client.");
+        }
     }
 
     /**
-     * Attempt token refresh using given access token
-     * When successful, persist new tokens
+     * Attempt token refresh using given refresh token
      * @param refreshToken - refresh token to use in request with token issuer
      * @returns tokens - Promise that resolves with new access and refresh tokens
      * @throws when token refresh fails
      */
-    public async doRefreshAccessToken(refreshToken: string): ReturnType<TokenRefreshFunction> {
+    public async doRefreshAccessToken(refreshToken: string): Promise<AccessTokens> {
         if (!this.inflightRefreshRequest) {
-            this.inflightRefreshRequest = this.getNewToken(refreshToken);
+            this.inflightRefreshRequest = this.getNewTokens(refreshToken);
         }
         try {
             const tokens = await this.inflightRefreshRequest;
-            await this.persistTokens(tokens);
             return tokens;
-        } catch (error) {
-            logger.error("Failed to refresh access token", error);
-            throw error;
         } finally {
             this.inflightRefreshRequest = undefined;
         }
     }
 
     /**
-     * Persist the new tokens after successfully refreshing
+     * Persist the new tokens
+     * Called after tokens are successfully refreshed
+     * This function is intended to be overriden by the consumer when persistence is necessary
```

Comment:
Again, be aware that when this is rendered, single line breaks are swallowed, so all these sentences get run together and are quite hard to parse. Please add some paragraph breaks, or at least full stops (preferably both), between sentences.